ctmds/commodity_price_series.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import pandas as pd
+
+from ctmds.country_datetime_series import get_country_datetime_series
+
+
+def get_commodity_price_series(for_date, country_code, granularity: str = "hourly", seed=None):
+    """
+    Generate synthetic price series for power, natural gas, and crude oil.
+
+    Parameters
+    ----------
+    for_date : str
+        Date in 'YYYY-MM-DD' format.
+    country_code : str
+        One of ['GB', 'FR', 'NL', 'DE'] representing the country code.
+    granularity : str, optional
+        Frequency of data points, one of ['hourly', 'half-hourly']. Default is 'hourly'.
+    seed : int or None, optional
+        Seed for the random number generator. Default is None.
+
+    Returns
+    -------
+    DataFrame
+        DataFrame with datetime index and price columns for power, natural gas, and crude oil.
+    """
+    # Create a datetime index
+    dates = get_country_datetime_series(for_date, country_code, granularity)
+    n = len(dates)
+    hours = dates.hour
+    days = dates.dayofyear
+
+    # Country-specific base prices (in €/MWh for power & gas, $/barrel for crude)
+    base_prices = {
+        "GB": {"power": 55, "natgas": 25, "crude": 72},
+        "FR": {"power": 50, "natgas": 20, "crude": 70},
+        "NL": {"power": 48, "natgas": 18, "crude": 69},
+        "DE": {"power": 45, "natgas": 22, "crude": 68},
+    }
+
+    # Get country-specific base prices
+    base_power_price = base_prices[country_code]["power"]
+    base_natgas_price = base_prices[country_code]["natgas"]
+    base_crude_price = base_prices[country_code]["crude"]
+
+    # Seasonal effect using sine wave
+    # Annual seasonality (365.25 days cycle)
+    # Peaks in January and July, troughs in April and October
+    annual_cycle = 4 * np.pi * (days - 365.25 / 2) / 365.25
```

Comment:
The countries we currently support are geographically close together and share similar climate. But what if we had Mexico or Australia here? How would this part of the code need to change? (Just a thought experiment - you don't need to implement. )